src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs

```diff
@@ -428,13 +433,7 @@ private static SyntaxNode GetLeftmostCondition(SyntaxNode node)
                         var declarators = ((LocalDeclarationStatementSyntax)node).Declaration.Variables;
                         // We require this to be the only declarator in the declaration statement
                         // to simplify definitive assignment check and the code fix for now
-                        var value = declarators.Count == 1 ? declarators[0].Initializer?.Value : null;
-                        if (value == null)
-                        {
-                            return null;
-                        }
-
-                        node = value;
+                        node = declarators.Count == 1 ? declarators[0].Initializer?.Value : null;
                         continue;
```

Comment:
what about having `break;` in all cases and a `return` from the default case? might be easier to understand in my opinion than having `continue` from inside a switch statement

Reply:
I would prefer to keep an escrow regression fix minimal, without style changes. Personally, I also prefer code as per your suggestion, but the current style is also readable as we loop with explicit continue statements and exit when we reach null node or node of unhandled kind.